src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/OptionFilter.php

```diff
@@ -0,0 +1,138 @@
+        $this->supportedAttributeTypes = $supportedAttributeTypes;
+        $this->supportedOperators = $supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeFilter(
+        AttributeInterface $attribute,
+        $operator,
+        $values,
+        $locale = null,
+        $channel = null,
+        $options = []
+    ) {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkLocaleAndChannel($attribute, $locale, $channel);
+
+        if (Operators::IS_EMPTY !== $operator && Operators::IS_NOT_EMPTY !== $operator) {
+            $this->checkValue($attribute, $values);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $channel);
+
+        switch ($operator) {
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_LIST:
+                $clause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Check if values are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $values
+     *
+     * @throws ObjectNotFoundException
+     */
+    protected function checkValue(AttributeInterface $attribute, $values)
+    {
+        FieldFilterHelper::checkArray($attribute->getCode(), $values, static::class);
+
+        foreach ($values as $value) {
+            FieldFilterHelper::checkIdentifier($attribute->getCode(), $value, static::class);
+        }
+
+        $options = $attribute->getOptions();
+
+        $expectedValues = [];
+        foreach ($options as $option) {
+            $expectedValues[] = $option->getCode();
+        }
+
+        foreach ($values as $value) {
```

Comment:
@jjanvier we did this check in [getIdsFromCodes](https://github.com/jjanvier/pim-community-dev/blob/TIP-613-unified/src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/OptionFilter.php#L78)